games-roguelike/nethack/nethack-3.6.1.ebuild

```diff
@@ -0,0 +1,121 @@
+}
+
+src_compile() {
+	append-cflags -I../include -DDLB -DSECURE -DTIMED_DELAY -DVISION_TABLES -DDUMPLOG -DSCORE_ON_BOTL
+	append-cflags '-DCOMPRESS=\"${EPREFIX}/bin/gzip\"' '-DCOMPRESS_EXTENSION=\".gz\"'
+	append-cflags "-DHACKDIR=\\\"${EPREFIX}/usr/$(get_libdir)/nethack\\\"" "-DVAR_PLAYGROUND=\\\"${EPREFIX}/var/games/nethack\\\""
+	append-cflags "-DDEF_PAGER=\\\"${PAGER}\\\""
+	append-cflags -DSYSCF "-DSYSCF_FILE=\\\"${EPREFIX}/etc/nethack.sysconf\\\""
+
+	use X && append-cflags -DX11_GRAPHICS -DUSE_XPM
+
+	LOCAL_MAKEOPTS=(
+		CC="$(tc-getCC)" CFLAGS="${CFLAGS}" LFLAGS="${LDFLAGS}"
+		WINTTYLIB="$($(tc-getPKG_CONFIG) --libs ncurses)"
+		HACKDIR="${EPREFIX}/usr/$(get_libdir)/nethack" INSTDIR="${ED}/usr/$(get_libdir)/nethack"
+		SHELLDIR="${ED}/usr/bin" VARDIR="${ED}/var/games/nethack"
+		)
+
+	emake "${LOCAL_MAKEOPTS[@]}" nethack recover Guidebook spec_levs
+
+	# Upstream still has some parallel compilation bugs
+	emake -j1 "${LOCAL_MAKEOPTS[@]}" all
+}
+
+src_install() {
+	emake "${LOCAL_MAKEOPTS[@]}" install
+
+	mv "${ED}/usr/$(get_libdir)/nethack/recover" "${ED}/usr/bin/recover-nethack" || die "Failed to move recover-nethack"
+
+	doman doc/nethack.6
+	newman doc/recover.6 recover-nethack.6
+	dodoc doc/Guidebook.txt
+
+	insinto /etc
+	newins sys/unix/sysconf nethack.sysconf
+
+	insinto /etc/skel
+	newins "${FILESDIR}/${PN}-3.6.0-nethackrc" .nethackrc
+
+	if use X ; then
+		cd "${S}/win/X11" || die "Failed to enter win/X11 directory"
+
+		mkdir -p "${ED}/etc/X11/app-defaults/" || die "Failed to make app-defaults directory"
+		mv "${ED}/usr/$(get_libdir)/nethack/NetHack.ad" "${ED}/etc/X11/app-defaults/" || die "Failed to move NetHack.ad"
+
+		newicon nh_icon.xpm nethack.xpm
+		make_desktop_entry ${PN} Nethack
+
+		# install nethack fonts
+		bdftopcf -o nh10.pcf nh10.bdf || die "Converting fonts failed"
+		bdftopcf -o ibm.pcf ibm.bdf || die "Converting fonts failed"
+		insinto "/usr/$(get_libdir)/nethack/fonts"
+		doins *.pcf
+		mkfontdir "${ED}/usr/$(get_libdir)/nethack/fonts" || die "mkfontdir failed"
+	fi
+
+	rm -r "${ED}/var/games/nethack" || die "Failed to clean var/games/nethack"
+	keepdir /var/games/nethack/save
+}
+
+pkg_preinst() {
+	enewgroup gamestat 36
+
+	fowners root:gamestat /var/games/nethack /var/games/nethack/save
+	fperms 2770 /var/games/nethack /var/games/nethack/save
+
+	fowners root:gamestat "/usr/$(get_libdir)/nethack/nethack"
+	fperms g+s "/usr/$(get_libdir)/nethack/nethack"
+}
+
+pkg_postinst() {
+	cd "${EROOT}/var/games/nethack" || die "Failed to enter ${EROOT}/var/games/nethack directory"
+
+	# Transition mechanism for <nethack-3.6.1 ebuilds. It's perfectly safe, so we'll just run it unconditionally.
+	chmod 2770 . save || die "Failed to chmod statedir"
+
+	# Those files can't be created earlier because we don't want portage to wipe them during upgrades
+	( umask 007 && touch logfile perm record xlogfile || die "Failed to create log files" )
+
+	touch -c bones* save/*  # non-critical
```

Comment:
Why do you need to do that?

Reply:
I don't remember the exact problem, but I think I added this to the 3.6.0 ebuild because nethack checks the timestamps of those files as a rudimentary anti-cheat mechanism.

Reply:
Hmm, i guess that's fine. Though I'd personally prefer if you used `nullglob` instead of relying on no-create with failed expansion. You'd have to expand the patterns into a local array, and call `touch` if it expanded to at least one filename. On using shopt, see this article: https://blogs.gentoo.org/mgorny/2016/01/25/mangling-shell-options-in-ebuilds/

Reply:
Oh, and adding comment why you're touching those files would be useful for future readers.

Reply:
That said, I can't find anything like that in the sources right now. I'll remove it and test if anything breaks.

Reply:
Ahh, I found it now. Nethack compares the mtimes of saves to the mtime of its executable as a compability guard. This made sense during the 3.4.3-->3.6.0 upgrade, but it's counterproductive for 3.6.* or rebuilds of the same version.